connectors/hub-concur/src/main/java/com/vmware/connectors/concur/service/HubConcurService.java

```diff
@@ -0,0 +1,232 @@
+import com.vmware.connectors.common.payloads.response.Cards;
+import com.vmware.connectors.common.utils.CardTextAccessor;
+import com.vmware.connectors.common.utils.CommonUtils;
+import com.vmware.connectors.concur.domain.ExpenseReportResponse;
+import com.vmware.connectors.concur.domain.PendingApprovalResponse;
+import com.vmware.connectors.concur.util.HubConcurUtil;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.util.HtmlUtils;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.text.NumberFormat;
+import java.util.Locale;
+
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.http.MediaType.APPLICATION_XML;
+
+@Service
+public class HubConcurService {
+
+    private static final Logger logger = LoggerFactory.getLogger(HubConcurService.class);
+
+    private final WebClient rest;
+    private final CardTextAccessor cardTextAccessor;
+    private final Resource concurRequestTemplate;
+    private final String serviceAccountAuthHeader;
+
+    @Autowired
+    public HubConcurService(
+            WebClient rest,
+            CardTextAccessor cardTextAccessor,
+            @Value("classpath:static/templates/concur-request-template.xml") Resource concurRequestTemplate,
+            @Value("${concur.service-account-auth-header}") String serviceAccountAuthHeader
+    ) {
+        this.rest = rest;
+        this.cardTextAccessor = cardTextAccessor;
+        this.concurRequestTemplate = concurRequestTemplate;
+        this.serviceAccountAuthHeader = serviceAccountAuthHeader;
+    }
+
+    public Mono<Cards> fetchCards(String baseUrl, Locale locale, String routingPrefix, HttpServletRequest request) {
+        logger.debug("fetchCards called: baseUrl={}, routingPrefix={}", baseUrl, routingPrefix);
+
+        return fetchAllRequests(baseUrl)
+                .flatMapMany(expenses -> Flux.fromIterable(expenses.getPendingApprovals()))
+                // TODO - do we need to filter on anything else?
+                .flatMap(expense -> fetchRequestData(baseUrl, expense.getId()))
+                .map(report -> makeCards(routingPrefix, locale, report, request))
+                .reduce(new Cards(), this::addCard);
+    }
+
+    private Mono<PendingApprovalResponse> fetchAllRequests(String baseUrl) {
+        String approver = "sdeswal@vmware-qa"; // TODO - change to user input
```

Comment:
Created APF-1714 to remove this hardcoded parameter from the hub concur connector.

Reply:
There is already a ticket for that.  I'm pretty sure it is APF-1544, but it might be another, check the Concur Connector epic.